app/scripts/models/reliers/supplicant.js

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+import { DEVICE_PAIRING_SCOPES } from '../../lib/constants';
+import OAuthErrors from '../../lib/oauth-errors';
+import OAuthRelier from './oauth';
+import Vat from '../../lib/vat';
+
+/*eslint-disable camelcase, sorting/sort-object-props*/
+const SENT_TO_AUTHORITY_FROM_SUPPLICANT_SCHEMA = {
+  access_type: Vat.accessType().renameTo('accessType'),
+  client_id: Vat.clientId().required().renameTo('clientId'),
+  code_challenge: Vat.codeChallenge().required().renameTo('codeChallenge'),
+  code_challenge_method: Vat.codeChallengeMethod().required().renameTo('codeChallengeMethod'),
+  keys_jwk: Vat.keysJwk().required().renameTo('keysJwk'),
+  response_type: Vat.string().required().valid('token', 'code').renameTo('responseType'),
+  redirect_uri: Vat.url().required().renameTo('redirectUri'),
+  scope: Vat.string().required().valid(DEVICE_PAIRING_SCOPES.join(' ')),
+  state: Vat.string().required()
+};
+
+var SUPPLICANT_QUERY_PARAM_SCHEMA = {
+  access_type: Vat.accessType().renameTo('accessType'),
+  client_id: Vat.clientId().required().renameTo('clientId'),
+  code_challenge: Vat.codeChallenge().required().renameTo('codeChallenge'),
+  code_challenge_method: Vat.codeChallengeMethod().required().renameTo('codeChallengeMethod'),
+  keys_jwk: Vat.keysJwk().required().renameTo('keysJwk'),
+  redirect_uri: Vat.url().required().renameTo('redirectUri'),
+  response_type: Vat.string().required().valid('token', 'code').renameTo('responseType'),
```

Comment:
@rfk should we remove it for the pairing here?

Reply:
Sorry, I'm not sure what "it" refers to here.  If it's the `redirect_uri` field, then yes, that seems reasonable to me on the Authority side.

Reply:
Sorry, `response_type`

Reply:
from mtg: `redirect_uri` can also probably go!

Reply:
@vladikoff this schema is the query param schema when loading the supplicant, I think it should stay here so we can redirect back to the app. The `SENT_TO_AUTHORITY_FROM_SUPPLICANT_SCHEMA` no longer includes either `redirect_uri` or `response_type`.